sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4456,6 +4456,20 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
         """.stripMargin),
       Seq(Row(2), Row(1)))
   }
+
+  test("SPARK-39448: Add ReplaceCTERefWithRepartition into nonExcludableRules list") {
```

Comment:
A question. Is it inevitable for us to have this?

As you know, we recommend `SQLQueryTestSuite` over `SQLQuerySuite` in these days, @wangyum ?

Reply:
I added a file:
https://github.com/apache/spark/blob/1ac77bb9ee496e8cdea67415bdb7cf839b9a4ac5/sql/core/src/test/resources/sql-tests/inputs/non-excludable-rule.sql